streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1200,37 +1198,28 @@ private String decorateTopic(final String topic) {
         return applicationId + "-" + topic;
     }
 
-    SubscriptionUpdates subscriptionUpdates() {
-        return subscriptionUpdates;
+    boolean usesPatternSubscription() {
+        return (!nodeToSourcePatterns.isEmpty());
+    }
+
+    synchronized Collection<String> sourceTopicCollection() {
+        log.debug("No source topics using pattern subscription found, using regular subscription for the main consumer.");
+
+        return sourceTopicNames;
     }
 
     synchronized Pattern sourceTopicPattern() {
+        log.debug("Found pattern subscribed source topics, falling back to pattern subscription for the main consumer.");
+
         if (topicPattern == null) {
-            final List<String> allSourceTopics = new ArrayList<>();
```

Comment:
This can be simplified by just using `sourceTopicNames`, which is identical to `nodeToSourceTopics.values()` but without the global topics, which we remove